APP-304: Add semantic design variables

wgardiner opened the pull request:
## Description
This PR adds custom Tailwind variables based on our new design system. 

Each client has a `tailwind.css` (`tailwind.css` for Regen, and `Terrasos.tailwind.css` for Terrasos), which defines all the CSS variables, `tailwind.common.js` then references them and exposes them for useage in Tailwind generated class names. This allows for the same variable to be defined per client, and the appropriate version referenced in based on the active client. 

Closes: [APP-304](https://regennetwork.atlassian.net/browse/APP-304?atlOrigin=eyJpIjoiODE0NjhlODFkZDQxNGI3ZWE5ZTBlYmVhMDQ2Njg1NjEiLCJwIjoiaiJ9
)
<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review. -->

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

<!-- This should include steps on how to test the fixes or features within the marketplace app and/or the website.
If applicable, this should also include links to the created/updated components on storybook. -->

1. Review Extend Colors in Storybook

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)


[APP-304]: https://regennetwork.atlassian.net/browse/APP-304?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

wgardiner commented:
The variables were generated by using the [Design Tokens](https://www.figma.com/community/plugin/888356646278934516/design-tokens) plugin for Figma. I also evaluated Token Studio. I found some issues with consistency in references and wrote a script to preprocess the output then run it through a custom [Style Dictionary](https://amzn.github.io/style-dictionary/#/) script. There was some manual work to map the generated CSS variables and structure to the app and Tailwind, but ideally we'd have a workflow that automates this process so that we can easily keep design and our app in sync. In the future we might consider having a source of truth that both Figma and our app read from, rather than having to export from Figma.

wgardiner commented:
I prefixed base colors, alias colors, and semantics with `bc`, `ac` and `s` respectively, just to prevent collisions during initial implementation, but we can easily update these based on feedback.

flagrede reviewed (commented):
Nice work!

Should we commit the script that you used to build the CSS variables from the Figma export ?

Also I saw that MUI is not updated here, is this going to be handled in a follow-up PR?

wgardiner commented:
I worked with Erika last week to get updated colors for `Terrasos.muiTheme.js` and `Terrasos.tailwind.css`. I also fixed and added scripts for generating theme file content from design tokens.

@flagrede are we already generating a storybook preview with the client .env variables set for Terrasos?

flagrede commented:
@wgardiner I don't think so no.

flagrede commented on design/.gitignore:
```diff
@@ -0,0 +1,176 @@
+build/
```
Was there an issue with the root `.gitignore` to add a second one?

flagrede commented on design/README.md:
````diff
@@ -0,0 +1,23 @@
+# Design
+
+This package processes [design tokens](https://tr.designtokens.org/format/) to generate CSS variables and a Tailwind config. It expects the design tokens exported from our Figma with the [Design Tokens plugin](https://www.figma.com/community/plugin/888356646278934516/design-tokens).
+
+```
+bun run build-tokens
+```
+
+It parses `tokens/tokens.json` and generates 
+- `base-regen.css`: base colors for Regen theme
+- `base-terrasos.css`: base colors for Terrasos theme
+- `alias-regen.css`: aliased colors that reference theme base colors (e.g. `primary.400` => `green.400`)
+- `alias-terrasos.css`: aliased colors that reference theme base colors
+- `semantic-regen-dark.css`: semantic colors for elements with a dark theme
+- `semantic-regen-light.css`: semantic colors for elements with a light theme
+- `semantic-terrasos-dark.css`: semantic colors for elements with a dark theme
+- `semantic-terrasos-light.css`: semantic colors for elements with a light theme
+- `tailwind.common.js`: a shared js file that can reference the relevant css variables for all theme clients
+
+These files can then be integrated into our app's theming system, by updating 
+- `tailwind.common.js`: shared interface
+- `tailwind.css`: default theme variables
+- `web-marketplace/src/clients/*/*.tailwind.css`: client specific theme variables
````
Great doc!

wgardiner replied:
Thanks!

flagrede reviewed (approved):
Looks good, nice work on this PR!
